recirq/qcqmc/trial_wf.py

```diff
@@ -0,0 +1,1584 @@
+    )
+
+    ansatz_circuit += heuristic_layer_circuit
+
+    superposition_circuit = (
+        cirq.Circuit(
+            [
+                cirq.H(fermion_index_to_qubit_map[10]),
+                cirq.CNOT(
+                    fermion_index_to_qubit_map[10], fermion_index_to_qubit_map[2]
+                ),
+                cirq.SWAP(
+                    fermion_index_to_qubit_map[2], fermion_index_to_qubit_map[12]
+                ),
+                cirq.CNOT(
+                    fermion_index_to_qubit_map[10], fermion_index_to_qubit_map[4]
+                ),
+                cirq.CNOT(
+                    fermion_index_to_qubit_map[12], fermion_index_to_qubit_map[0]
+                ),
+                cirq.SWAP(fermion_index_to_qubit_map[4], fermion_index_to_qubit_map[8]),
+                cirq.SWAP(
+                    fermion_index_to_qubit_map[0], fermion_index_to_qubit_map[14]
+                ),
+            ]
+        )
+        + ansatz_circuit
+    )
+
+    ansatz_circuit = (
+        cirq.Circuit(
+            [
+                cirq.X(fermion_index_to_qubit_map[8]),
+                cirq.X(fermion_index_to_qubit_map[10]),
+                cirq.X(fermion_index_to_qubit_map[12]),
+                cirq.X(fermion_index_to_qubit_map[14]),
+            ]
+        )
+        + ansatz_circuit
+    )
+
+    return superposition_circuit, ansatz_circuit
+
+
+def get_circuits(
+    *,
+    two_body_params: np.ndarray,
+    # from wf_params:
+    n_orb: int,
+    n_elec: int,
+    heuristic_layers: Tuple[LayerSpec, ...],
+) -> Tuple[cirq.Circuit, cirq.Circuit]:
+    """A function that runs a specialized method to get the ansatz circuits."""
+
+    # TODO(?): Just input one of these quantities.
+    if n_orb != n_elec:
+        raise ValueError("n_orb must equal n_elec.")
+
+    circ_funcs = {
+        2: get_4_qubit_pp_circuits,
+        4: get_8_qubit_circuits,
+        6: get_12_qubit_circuits,
+        8: get_16_qubit_circuits,
+    }
+    try:
+        circ_func = circ_funcs[n_orb]
+    except KeyError:
+        raise NotImplementedError(f"No circuits for n_orb = {n_orb}")
+
+    return circ_func(
+        two_body_params=two_body_params,
+        n_elec=n_elec,
+        heuristic_layers=heuristic_layers,
+    )
+
+
+def get_two_body_params_from_qchem_amplitudes(
+    qchem_amplitudes: np.ndarray,
+) -> np.ndarray:
+    """Translates perfect pairing amplitudes from qchem to rotation angles.
+
+    qchem style: 1 |1100> + t_i |0011>
+    our style: cos(\theta_i) |1100> + sin(\theta_i) |0011>
+    """
+
+    two_body_params = np.arccos(1 / np.sqrt(1 + qchem_amplitudes**2)) * np.sign(
+        qchem_amplitudes
+    )
+
+    # Numpy casts the array improperly to a float when we only have one parameter.
+    two_body_params = np.atleast_1d(two_body_params)
+
+    return two_body_params
+
+
+#################### Here be dragons.###########################################
```

Comment:
WUT?

Reply:
Removed post refactor (I think this is a vestige of the private repo)